python/ray/tune/trial_executor.py

```diff
@@ -177,7 +187,19 @@ def force_reconcilation_on_next_step_end(self):
     def on_no_available_trials(self, trial_runner):
         if self._queue_trials:
             return
-        for trial in trial_runner.get_trials():
+
+        all_trials = trial_runner.get_trials()
+        all_trials_are_pending = all(
+            trial.status == Trial.PENDING for trial in all_trials)
+        if all_trials_are_pending and not is_ray_cluster() and (
+                trial_runner.iteration +
```

Comment:
I was hoping to find a EVERY_N equivalent for python logging module but can't find any. Open to suggestions if another API to TrialRunner to get iteration count can be saved :)

Reply:
Maybe record the latest iteration when we raised the message?

Reply:
Please also check if we run in this message very often - does 100 iterations mean we print the message every 10 seconds (because of our 0.1s timeout)?

Reply:
Perhaps it would be better to use time instead of iterations altogether?

Reply:
> Please also check if we run in this message very often - does 100 iterations mean we print the message every 10 seconds (because of our 0.1s timeout)?
Kai, what 0.1s timeout you are referring to? Is this it? https://github.com/ray-project/ray/blob/master/python/ray/tune/trial_runner.py#L552. I don't think this code path is even exercised in this case, as it's gated by get_running_trials() returning some trial to proceed (None in our case).
So I think you are right - 100 iterations is probably too frequent. 

To Antoni's point, it's probably better to just use time. And we should pick up something that is meaningful - some time that user is just about to wonder what our code is doing. How about 5 seconds?

Reply:
5 seconds sounds good. We should probably make it an env var too, so users can change/disable it if they want to

Reply:
> 5 seconds sounds good. We should probably make it an env var too, so users can change/disable it if they want to

Thinking loud here. If we make it an env variable, wouldn't that mean user needs to explicitly set it in their set up commands to take benefit of this warning msg? I would consider these users already more aware of this issue to the point of even setting this in their yaml file. The general users are still likely to just get stuck without hints about what is happening. Thoughts?

Reply:
We can have default values for env variables - we'd just set it to 5 seconds